LandSandBoat/server
# [core] Adds damage cap mod

**_I affirm:_**

## What does this pull request do?
Adds damage cap mod for mobs that have set values for damage they take.

There are several pieces of content that have "damage caps" in the game. Some examples are: Crustacean Conundrum bcnm, Suttung (dyna-qufim), Antaeus (dyna-qufim) and colonization reives.

Unfortunately we do not know or understand EXACTLY how these damage caps works and the amount of research required to fully understand the system is out of reach for a long long time. Nonetheless here is my proposal for the damage mod cap.

Lets use Dynamis-Qufim Suttung as an example. From Siknoz retail capture it has been viewed that the mob takes 99 capped damage but anything over 99 seems to have a "random" factor between a certain variant (observed to be roughly 10).  We do not know or understand the exact mechanic behind this as it can have 100 different factors.

### The proposal:
Implement two mods. `RECEIVED_DAMAGE_CAP` and `RECEIVED_DAMAGE_VARIANT`
By having these two mods we can closely replicate the observed damage seen. By setting the DAMAGE_CAP to 99 and the DAMAGE_VARIANT to 9 the damage observed will be between 90 and 99 is the damage is above 99.

For cases like Crustacean Conundrum bcnm where the damage cap appears to be 2 or for Reives if you don't meet the requirements then the damage is 0 or 1 this will also work. By setting the DAMAGE_CAP to 1 and the DAMAGE_VARIANT to 1 the output will generate either a 0 or a 1.

Things I tested:
damage spells, abilities (jump, chi blast, EES, etc.), bluemagic, bio/dia, summons, pets, physical damage, ranged damage, magic damage, breathe damage



## Steps to test these changes
Go to any mob you want.
!setmod RECEIVED_DAMAGE_CAP 100
!setmod RECEIVED_DAMAGE_VARIANT 10
Hit the mob for over 100 damage and see the damage random between 90-100

!setmod RECEIVED_DAMAGE_CAP 1
!setmod RECEIVED_DAMAGE_VARIANT 1
Hit the mob for over 1 damage and see the damage random between 0-1
